### ISSUE TITLE
Hubot 3.0.1 robot.messageRoom with HipChat Adapter throws

### ISSUE THREAD

**Issue by mdouglass** [hubotio/hubot#1386]
The following code when run using the HipChat adapter throws an error.
```javascript
robot.router.get('/hubot/test', (_req, res) => {
  res.status(200).json({})
  robot.messageRoom('119062_bot_testing@conf.hipchat.com', 'This is a test message')
})
```

Error:
```
TypeError: Cannot read property 'message' of undefined
  at Robot.HipChat.send (/opt/kixeye/20170815-235000000/node_modules/hubot-hipchat/src/hipchat.coffee:35:7, <js>:50:37)
  at Robot.messageRoom (/opt/kixeye/20170815-235000000/node_modules/hubot/src/robot.js:608:23)
  at robot.router.get (/opt/kixeye/20170815-235000000/scripts/router.js:13:11)
  at callbacks (/opt/kixeye/20170815-235000000/node_modules/express/lib/router/index.js:164:37)
  at param (/opt/kixeye/20170815-235000000/node_modules/express/lib/router/index.js:138:11)
  at pass (/opt/kixeye/20170815-235000000/node_modules/express/lib/router/index.js:145:5)
  at Router._dispatch (/opt/kixeye/20170815-235000000/node_modules/express/lib/router/index.js:173:5)
  at Object.router (/opt/kixeye/20170815-235000000/node_modules/express/lib/router/index.js:33:10)
  at next (/opt/kixeye/20170815-235000000/node_modules/connect/lib/proto.js:174:15)
  at Object.multipart [as handle] (/opt/kixeye/20170815-235000000/node_modules/connect-multiparty/index.js:47:63)
  at next (/opt/kixeye/20170815-235000000/node_modules/connect/lib/proto.js:174:15)
  at Object.urlencodedParser [as handle] (/opt/kixeye/20170815-235000000/node_modules/body-parser/lib/types/urlencoded.js:88:40)
  at next (/opt/kixeye/20170815-235000000/node_modules/connect/lib/proto.js:174:15)
  at Object.jsonParser [as handle] (/opt/kixeye/20170815-235000000/node_modules/body-parser/lib/types/json.js:100:40)
  at next (/opt/kixeye/20170815-235000000/node_modules/connect/lib/proto.js:174:15)
  at Object.query [as handle] (/opt/kixeye/20170815-235000000/node_modules/connect/lib/middleware/query.js:43:5)
  at next (/opt/kixeye/20170815-235000000/node_modules/connect/lib/proto.js:174:15)
  at Object.app.use [as handle] (/opt/kixeye/20170815-235000000/node_modules/hubot/src/robot.js:436:14)
  at next (/opt/kixeye/20170815-235000000/node_modules/connect/lib/proto.js:174:15)
  at Object.expressInit [as handle] (/opt/kixeye/20170815-235000000/node_modules/express/lib/middleware.js:30:5)
  at next (/opt/kixeye/20170815-235000000/node_modules/connect/lib/proto.js:174:15)
  at Object.query [as handle] (/opt/kixeye/20170815-235000000/node_modules/connect/lib/middleware/query.js:43:5)
  at next (/opt/kixeye/20170815-235000000/node_modules/connect/lib/proto.js:174:15)
  at Function.app.handle (/opt/kixeye/20170815-235000000/node_modules/connect/lib/proto.js:182:3)
  at Server.app (/opt/kixeye/20170815-235000000/node_modules/connect/lib/connect.js:67:37)
  at emitTwo (events.js:125:13)
  at Server.emit (events.js:213:7)
  at parserOnIncoming (_http_server.js:602:12)
  at HTTPParser.parserOnHeadersComplete (_http_common.js:116:23)
```

**Comment by mdouglass** [comment]
I believe this is happening because of the code at: https://github.com/hubotio/hubot/blob/ec8fe150e5d4140218e1a9f0e5461d14b16c1f57/src/robot.js#L608

If I change the this parameter in the apply call to robot.adapter then the messageRoom call succeeds.

```javascript
  robot.router.get('/hubot/test', (_req, res) => {
    res.status(200).json({})
    function messageRoom(room/* , ...strings */) {
      const strings = [].slice.call(arguments, 1)
      const envelope = { room }
      robot.adapter.send.apply(robot.adapter, ([envelope].concat(strings)))
    }
    messageRoom('119062_bot_testing@conf.hipchat.com', 'This is a test message')
  })
```

**Comment by liorbrauer** [comment]
Re-opening this since the issue has not been fixed in the latest version 3.0.1 (although PR #1390 has been merged to master).
